### PR TITLE
Fix follow_links() loop and lookup()

### PR DESCRIPTION
**Pull request by jorgebay**
Fixes infinite loop in `follow_links()` method and fixes `lookup()` to only return an `Option<EntryKey>` (instead of wrapped in a result).

### CODE REVIEW

File: common/fs/src/cache/mod.rs [hunk 1]
```diff
@@ -849,65 +849,53 @@ where
...
-                        continue;
-                    }
+            parent = self.follow_links(parent, _entries).and_then(|e| {
+                if let Some(e) = _entries.get(e) {
+                    e.borrow()
+                        .children()?
+                        .get(&component)
+                        .and_then(|entry| self.follow_links(*entry, _entries))
+                } else {
+                    error!("Failed to find entry");
```
Comment (c-nixon) [inline common/fs/src/cache/mod.rs:875]: Is this always an error? Or should it be a warn/info?

File: common/fs/src/tail.rs [hunk 2]
```diff
@@ -204,7 +204,7 @@ impl Tailer {
                             // if Some(..) = entries.get(key) clauses)
                             let mut entries = fs.entries.borrow_mut();
                             if entries.remove(entry_ptr).is_some() {
-                                debug!(
+                                info!(
```
Comment (c-nixon) [inline common/fs/src/tail.rs:207]: Should this be put back to debug? If we'd like to keep it at info the 
message should be tweaked to not talk about the map, as that's an implementation detail that doesn't need logged

**Review by c-nixon — COMMENTED** [review]
Awesome, great catch.

2 small queries around the logging, but the change looks solid

File: common/fs/src/cache/mod.rs (hunk 1 above)
Comment (jorgebay) [reply common/fs/src/cache/mod.rs:875]: Good point, I'll change it to info.

File: common/fs/src/tail.rs (hunk 2 above)
Comment (jorgebay) [reply common/fs/src/tail.rs:207]: I see what you mean, having it an info level can be useful but the message doesn't make much sense for the user, how about: `"Removed file information, currently tracking {} files"`

Comment (c-nixon) [reply common/fs/src/tail.rs:207]: That sounds good to me!

**Review by c-nixon — APPROVED** [review]

**Pull request merged**